service/policy/kasregistry/key_access_server_registry.proto

```diff
@@ -160,36 +230,61 @@ message ListKeyAccessServerGrantsResponse {
 }
 
 service KeyAccessServerRegistryService {
-  rpc ListKeyAccessServers(ListKeyAccessServersRequest) returns (ListKeyAccessServersResponse) {
-    option (google.api.http) = {get: "/key-access-servers"};
+  rpc ListKeyAccessServers(ListKeyAccessServersRequest)
+      returns (ListKeyAccessServersResponse) {
+    option (google.api.http) = {
+      get : "/key-access-servers"
+    };
     option idempotency_level = NO_SIDE_EFFECTS;
   }
 
-  rpc GetKeyAccessServer(GetKeyAccessServerRequest) returns (GetKeyAccessServerResponse) {
-    option (google.api.http) = {get: "/key-access-servers/{id}"};
+  rpc GetKeyAccessServer(GetKeyAccessServerRequest)
+      returns (GetKeyAccessServerResponse) {
+    option (google.api.http) = {
+      get : "/key-access-servers/{id}"
+    };
     option idempotency_level = NO_SIDE_EFFECTS;
   }
 
-  rpc CreateKeyAccessServer(CreateKeyAccessServerRequest) returns (CreateKeyAccessServerResponse) {
+  rpc CreateKeyAccessServer(CreateKeyAccessServerRequest)
+      returns (CreateKeyAccessServerResponse) {
     option (google.api.http) = {
-      post: "/key-access-servers"
-      body: "*"
+      post : "/key-access-servers"
+      body : "*"
     };
   }
 
-  rpc UpdateKeyAccessServer(UpdateKeyAccessServerRequest) returns (UpdateKeyAccessServerResponse) {
+  rpc UpdateKeyAccessServer(UpdateKeyAccessServerRequest)
+      returns (UpdateKeyAccessServerResponse) {
     option (google.api.http) = {
-      patch: "/key-access-servers/{id}"
-      body: "*"
+      patch : "/key-access-servers/{id}"
+      body : "*"
     };
   }
 
-  rpc DeleteKeyAccessServer(DeleteKeyAccessServerRequest) returns (DeleteKeyAccessServerResponse) {
-    option (google.api.http) = {delete: "/key-access-servers/{id}"};
+  rpc DeleteKeyAccessServer(DeleteKeyAccessServerRequest)
+      returns (DeleteKeyAccessServerResponse) {
+    option (google.api.http) = {
+      delete : "/key-access-servers/{id}"
+    };
   }
 
-  rpc ListKeyAccessServerGrants(ListKeyAccessServerGrantsRequest) returns (ListKeyAccessServerGrantsResponse) {
-    option (google.api.http) = {get: "/key-access-servers/grants"};
+  rpc ListKeyAccessServerGrants(ListKeyAccessServerGrantsRequest)
+      returns (ListKeyAccessServerGrantsResponse) {
+    option (google.api.http) = {
+      get : "/key-access-servers/grants"
+    };
     option idempotency_level = NO_SIDE_EFFECTS;
   }
+
+  rpc CreateKey(CreateKeyRequest) returns (CreateKeyResponse) {}
+
+  rpc GetKey(GetKeyRequest) returns (GetKeyResponse) {}
+
+  rpc ListKeys(ListKeysRequest) returns (ListKeysResponse) {}
+
+  rpc UpdateKey(UpdateKeyRequest) returns (UpdateKeyResponse) {}
+
+  // DeleteKey is a soft delete, marking the key as inactive.
+  rpc DeleteKey(DeleteKeyRequest) returns (DeleteKeyResponse) {}
```

Comment:
Can we please update this to `DeactivateKey` like the other safe deletion/deactivations of policy objects that are not deletions (because the unique constraints remain enforced by the db when inactive to prevent data leakage)?